app/frontend/src/features/messages/groupchats/groupChatsSlice.ts

```diff
@@ -0,0 +1,123 @@
+import { flow, getParent, Instance, types } from "mobx-state-tree";
+import { GroupChat, Message } from "../../../pb/conversations_pb";
+import { service } from "../../../service";
+
+export const initialState = {
+  groupChats: [],
+  error: "",
+  loading: false,
+  groupChatView: {
+    groupChat: null,
+    messages: [],
+    error: "",
+    loading: false,
+  },
+};
+
+export const GroupChatsModel = types
+  .model({
+    groupChats: types.array(types.frozen<GroupChat.AsObject>()),
+    error: types.string,
+    loading: types.boolean,
+    groupChatView: types
+      .model({
+        error: types.string,
+        loading: types.boolean,
+        messages: types.array(types.frozen<Message.AsObject>()),
+        groupChat: types.maybeNull(types.frozen<GroupChat.AsObject>()),
+      })
+      .actions((self) => {
+        const fetchMessages = flow(function* (groupChat) {
```

Comment:
I personally love generator functions, but this does raise the learning curve

Reply:
In this particular usage of generator functions it is easy: write `yield` where you would write `await` and start with `flow(function*`
https://mobx-state-tree.js.org/concepts/async-actions

Reply:
Aren't async-await additional syntactic sugar on top of generators? I never needed to use them apart from libraries that mandate them like mobx here or things like redux saga (which I haven't really used before either)

Reply:
Yes indeed, syntactical sugar. But without it, it would be much harder for many people to write async code. I also have done Saga for some time long ago. It was even more complex than Thunks.